File: soccer_field_reservations/bar-service/src/main/java/br/com/football/auth/api/AuthController.java
package br.com.football.auth.api;

import br.com.football.auth.domain.entity.Usuario;
import br.com.football.auth.domain.repository.UsuarioRepository;
import br.com.football.auth.security.JwtService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.*;

import java.util.Map;
import java.util.Optional;

@RestController
@RequestMapping("/auth")
public class AuthController {

    private final UsuarioRepository usuarioRepository;
    private final JwtService jwtService;
    private final PasswordEncoder passwordEncoder;

    public AuthController(
            UsuarioRepository usuarioRepository,
            JwtService jwtService,
            PasswordEncoder passwordEncoder
    ) {
        this.usuarioRepository = usuarioRepository;
        this.jwtService = jwtService;
        this.passwordEncoder = passwordEncoder;
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody LoginRequest request) {

        Optional<Usuario> optionalUsuario =
                usuarioRepository.findByUsername(request.username());

        if (optionalUsuario.isEmpty()) {
            return ResponseEntity
                    .status(HttpStatus.UNAUTHORIZED)
                    .body(Map.of("message", "Usuário ou senha inválidos"));
        }

        Usuario usuario = optionalUsuario.get();

        // Validação segura com BCrypt
        if (!passwordEncoder.matches(request.password(), usuario.getPassword())) {
            return ResponseEntity
                    .status(HttpStatus.UNAUTHORIZED)
                    .body(Map.of("message", "Usuário ou senha inválidos"));
        }

        String token = jwtService.gerarToken(
                String.valueOf(usuario.getId()),
                usuario.getUsername(),
                usuario.getRole().name()
        );

        return ResponseEntity.ok(
                Map.of(
                        "token", token,
                        "role", usuario.getRole().name()
                )
        );
    }

    // =========================
    // DTO interno simples
    // =========================
    public record LoginRequest(
            String username,
            String password
    ) {}
}

File: soccer_field_reservations/backend/bin/main/com/example/soccer/controller/AuthController.kt
package com.example.soccer.controller

import com.example.soccer.service.AuthService
import org.springframework.http.ResponseEntity
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.web.bind.annotation.*

data class LoginRequest(val login: String, val senha: String)
data class ChangePasswordRequest(val oldPass: String, val newPass: String)

@RestController
@RequestMapping("/api/auth")
class AuthController(private val authService: AuthService) {

    @PostMapping("/login")
    fun login(@RequestBody req: LoginRequest): ResponseEntity<Any> {
        return try {
            val token = authService.login(req.login, req.senha)
            ResponseEntity.ok(mapOf("token" to token, "success" to true))
        } catch (e: Exception) {
            ResponseEntity.status(401).body(mapOf("success" to false, "message" to e.message))
        }
    }

    @PostMapping("/change-password")
    fun changePassword(@RequestBody req: ChangePasswordRequest): ResponseEntity<Any> {
        return try {
            val principal = SecurityContextHolder.getContext().authentication.principal as UserDetails
            authService.changePassword(principal.username, req.oldPass, req.newPass)
            ResponseEntity.ok(mapOf("success" to true, "message" to "Senha alterada com sucesso"))
        } catch (e: Exception) {
            ResponseEntity.badRequest().body(mapOf("success" to false, "message" to e.message))
        }
    }
}
